### DISCUSSION TITLE
How to get access to Serial Console of the microVM when started with jailer?

### DISCUSSION THREAD

**Discussion by chwzr** [firecracker-microvm/firecracker#4083] [Q&A] [answered]
I am starting firecracker vms with the jailer (with demonize and config.json flags:  `["--daemonize", "--", "--config-file", "./config.json" ]`.

How to get access to the serial console then?

when starting firecracker standalone its mapped to stdin/stdout of the firecracker process. but how to attach to the console when started with the jailer?

already tested to write to / read from `/proc/<firecracker-pid>/fd/0` and `/proc/<firecracker-pid>/fd/1` but this is not working.

**Comment by kalyazin** [comment, marked as answer]
Hi @chwzr . Thank you for your question. Daemonization is one of security measures that harden microVM execution and it indeed prohibits direct operator access to the guest, so it is expected that stdin and stdout are not accessible.
